lib/y-richtext.coffee

```diff
@@ -45,37 +72,26 @@ class RichText extends BaseClass
   _setModel: (model) ->
     super
 
-    noneFound = true
-    for cursor in (@_get "cursors").val()
-      if cursor.author == @author
-        @updateCursorPosition(cursor.position)
-        noneFound = false
-        break
-
-    if noneFound
-      @_setCursor @editor.getCursorPosition()
-
-    delete @_characters
-    delete @_selections
-    delete @_cursors
+  _name: "RichText"
 
-  # insert our own cursor in the cursors list
+  # insert our own cursor in the cursors object
   # @param position [Integer] the position where to insert it
-  setCursor = (position) ->
-    character = (@_get "characters").val(position)
-    selfCursor =
-      author: @author
-      position: character
-      color: "grey" # FIXME
-    (@_get "cursors").insert 0, selfCursor
-    @selfCursor = (@_get "cursors").ref 0
+  setCursor : (position) ->
+    @selfCursor = (@_get "characters").ref(position)
+    (@_get "cursors").val(@_model.HB.getUserId(), @selfCursor)
+
 
   # pass deltas to the character instance
   # @param deltas [Array<Object>] an array of deltas (see ot-types for more info)
-  passDeltas = (deltas) ->
+  passDeltas : (deltas) => # TODO: don't bind to $this
+    if this.lock_editor_propagation
+      # break, if lock is on
+      return
```

Comment:
This won't work because is another delta is passed while this one is being processed, the delta will just disappear.
You should wait for the lock to be lifted instead.